test/distributed/org/apache/cassandra/distributed/mock/nodetool/InternalNodeProbe.java

```diff
@@ -66,26 +63,9 @@ protected void connect()
         mbeanServerConn = null;
         jmxc = null;
 
-        if (withNotifications)
-        {
-            ssProxy = StorageService.instance;
-        }
-        else
-        {
-            // replace the notification apis with a no-op method
-            StorageServiceMBean mock = Mockito.spy(StorageService.instance);
-            Mockito.doNothing().when(mock).addNotificationListener(Mockito.any(), Mockito.any(), Mockito.any());
-            try
-            {
-                Mockito.doNothing().when(mock).removeNotificationListener(Mockito.any(), Mockito.any(), Mockito.any());
-                Mockito.doNothing().when(mock).removeNotificationListener(Mockito.any());
-            }
-            catch (ListenerNotFoundException e)
-            {
-                throw new AssertionError(e);
-            }
-            ssProxy = mock;
-        }
+        StorageService.instance.skipNotificationListeners = !withNotifications;
```

Comment:
pretty simple, right? then we change it back to default (false) in close method.